DMCompiler/DM/Visitors/DMVisitorObjectBuilder.cs

```diff
@@ -105,7 +107,26 @@ class DMVisitorObjectBuilder : DMASTVisitor {
                     dmObject = DMObjectTree.GetDMObject(_currentObject.Path.Combine(procDefinition.ObjectPath.Value));
                 }
 
-                if (!procDefinition.IsOverride && dmObject.HasProc(procName)) {
+                if (procDefinition.IsOverride)
+                {
+                    if (dmObject is not null)
+                    {
+                        var parentProcs = dmObject.GetProcs(procName);
+                        if (parentProcs is not null && parentProcs.Count >= 1)
+                        {
+                            var parent = parentProcs[dmObject.GetProcs(procName).Count - 1];
+                            if (procDefinition.ReturnTypes.Equals(DMValueType.Anything))
```

Comment:
So this would cause the following code to error?
```
proc/return_num() as num
	return 9

return_num() as anything
	return "A"
```
I personally think this is a little confusing and shouldn't be allowed.